rest-api-spec/src/main/resources/rest-api-spec/test/search/30_limits.yml

```diff
@@ -26,14 +26,49 @@ setup:
         from: 10000
 
 ---
-"Request with negative from value":
+"Request with negative from value url parameter":
+
+  - skip:
+      version: " - 7.9.99"
+      reason: waiting for backport
+
   - do:
-      catch:      /\[from\] parameter cannot be negative/
+      search:
+        rest_total_hits_as_int: true
+        index: test_1
+        # value of -1 should be accepted as unset value
+        from: -1
+
+  - do:
+      catch:      /\[from\] parameter cannot be negative but was \[-2\]/
       search:
         rest_total_hits_as_int: true
         index: test_1
         from: -2
 
+---
+"Request with negative from value in body":
+
+  - skip:
+      version: " - 7.9.99"
+      reason: waiting for backport
+
+  - do:
+      search:
+        rest_total_hits_as_int: true
+        index: test_1
+        # value of -1 should be accepted as unset value
```

Comment:
Can we try to address this ? For instance we could set the default value directly rather than delaying this to the creation of the search context ?

Reply:
I'd be open to forbid users setting from to  "-1" entirely on 8.0 but since we currently accept it in the request body (and treat it as 0 later) I think we should allow it at least on 7.x to not break existing queries there. Every other negative value should alread throw a different exception, so thats fine I think. We can log a warning on 7.x when using "-1" to prepare for the change in 8.0 if thats okay?